From 6bcb4831a4dc15a7de2b98c4141ff357111e0d9d Mon Sep 17 00:00:00 2001
From: Du Yilong <duyilong@wxiat.com>
Date: Mon, 25 Jul 2022 14:24:36 +0800
Subject: [PATCH] sw64: kvm: expand the number of SWVM_IRQS
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 6bcb4831a4dc15a7de2b98c4141ff357111e0d9d
Modified-by-SEL: No


Sunway inclusion
category: bugfix
bugzilla: https://gitee.com/openeuler/kernel/issues/I5PN9S

--------------------------------

As more devices are hot-plugged in a guest os, it will fail to respond
interrupts because of insufficient interrupt resource and appear to be
stuck.

To fix this issue, expand the maximum number of irq supported to 256.

Signed-off-by: Du Yilong <duyilong@wxiat.com>
Signed-off-by: Gu Zitao <guzitao@wxiat.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 arch/sw_64/include/uapi/asm/kvm.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/sw_64/include/uapi/asm/kvm.h b/arch/sw_64/include/uapi/asm/kvm.h
index ff1b6e7f096f..126c2a1d7411 100644
--- a/arch/sw_64/include/uapi/asm/kvm.h
+++ b/arch/sw_64/include/uapi/asm/kvm.h
@@ -5,7 +5,7 @@
 /*
  * KVM SW specific structures and definitions.
  */
-#define SWVM_IRQS 64
+#define SWVM_IRQS 256
 enum SW64_KVM_IRQ {
 	SW64_KVM_IRQ_IPI = 27,
 	SW64_KVM_IRQ_TIMER = 9,
-- 
2.33.0

